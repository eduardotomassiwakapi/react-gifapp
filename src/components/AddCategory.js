import React, { useState } from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({ setCategories, /*limit, setLimit*/ }) => {

    const [inputCategoryValue, setInputCategoryValue] = useState('');
    //const [inputLimitValue, setInputLimitValue] = useState(limit)
    
    const handleCategoryInputChange = (e) => {
        //console.log("handleCategoryInputChange="+e.target.value);
        setInputCategoryValue(e.target.value);
    }
    // const handleLimitInputChange = (e) => {
    //     setInputLimitValue(e.target.value);
    // }
    const handleSubmit = (e) => {
        e.preventDefault();
        if (inputCategoryValue.trim().length > 2) {
            //console.log("handleSubmit="+inputCategoryValue);
            setCategories(oldCategories => [inputCategoryValue, ...oldCategories,]);
            //setLimit(inputLimitValue);
            setInputCategoryValue('');
        }
    }


    return (
        <form onSubmit={handleSubmit}>
            <input
                type="text" 
                value={inputCategoryValue}
                onChange={ (e) => handleCategoryInputChange(e)} />
            {/* <input type="number" 
            value={limit} onChange={ (e) => handleLimitInputChange(e)} /> */}
        </form>
    )
};

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired,
   //setLimit: PropTypes.func.isRequired
};
