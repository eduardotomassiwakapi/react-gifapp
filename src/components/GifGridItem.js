import React from 'react'
import PropTypes from 'prop-types'

export const GifGridItem = ({title,url}) => {
    //console.log({id,title,url});
    return (
        <div className='card animate__animated animate__fadeIn'>
            <h4>{title}</h4>
            <img src={url} alt={title} />
        </div>
    )
}

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
};