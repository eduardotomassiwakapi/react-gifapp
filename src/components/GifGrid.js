// import React, { useState, useEffect } from 'react'
// import { fetchGifs } from '../helpers/fetchGifs';
import { GifGridItem } from './GifGridItem';

import { useFetchGifs } from "../hooks/useFetchGifs"

export const GifGrid = ({ category, limit }) => {

    const apiKey = "1H3c5Ybz7bYQgUPoGaTiMK228pvRc0nK";
    const apiUrl = `https://api.giphy.com/v1/gifs/search?rating=r&q=${encodeURI(category)}&limit=${limit}&api_key=${apiKey}`;
    const {data: images, loading} = useFetchGifs(apiUrl);

    return ( 
        <>
            <h3>{category}</h3>
            {loading && <p className="animate__animated animate__flash">Loading...</p>}
            <div className="card-grid">
                {
                    images.map((img) => (
                        <GifGridItem
                            key={img.id}
                            {...img}
                        />
                    ))
                }
            </div>
        </>
    )
}
