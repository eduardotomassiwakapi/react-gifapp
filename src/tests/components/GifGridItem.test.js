import { shallow } from 'enzyme'
import { GifGridItem } from "../../components/GifGridItem";

describe('<GifGridItem />', () => {

    const title = "title";
    const url = 'http//:someulr.com';
    
    // BEFORE TEST
    let wrapper;
    beforeEach(() => {
        wrapper = shallow(<GifGridItem title={title} url={url} />);
    })

    test('Should render properly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('should render with h4 and img', () => {
        const h4 = wrapper.find('h4');
        expect(h4.text()).toBe(title);
        const img = wrapper.find('img').at(0);
        //console.log(img.props());
        expect(img.prop("src")).toBe(url);
        expect(img.prop("alt")).toBe(title);
    })

    test('div should have css class', () => {
        const div = wrapper.find('div').props();
        // console.log(div);
        expect(div.className).toBe('card animate__animated animate__fadeIn');
        expect(div.className.includes('animate__animated')).toBe(true);
        //expect(div).toHaveClass('card animate__animated animate__fadeIn') wont work cause it has className
    })

})
