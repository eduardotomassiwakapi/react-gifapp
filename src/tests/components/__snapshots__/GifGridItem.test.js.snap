// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifGridItem /> Should render properly 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <h4>
    title
  </h4>
  <img
    alt="title"
    src="http//:someulr.com"
  />
</div>
`;
